digid_eherkenning/saml2/eherkenning.py

```diff
@@ -268,19 +269,17 @@ def create_classifiers_element(classifiers: list) -> ElementMaker:
     return ESC("Classifiers", *classifiers_elements)
 
 
-def create_key_descriptor(x509_certificate_content):
-    x509_certificate = crypto.load_certificate(
-        crypto.FILETYPE_PEM, x509_certificate_content
-    )
-    key_descriptor_cert = b64encode(
-        crypto.dump_certificate(crypto.FILETYPE_ASN1, x509_certificate)
-    ).decode("ascii")
-
-    certificate = x509_certificate.to_cryptography()
+def create_key_descriptor(x509_certificate_content: bytes):
+    certificate = load_pem_x509_certificate(x509_certificate_content)
     key_name = binascii.hexlify(
         certificate.fingerprint(certificate.signature_hash_algorithm)
     ).decode("ascii")
 
+    # grab the actual base64 data describding the certificate, but without the
+    # BEGIN/END CERTIFICATE headers and footers and stripped of line breaks.
```

Comment:
Is this new behaviour or is it the same as before?

Reply:
Same as before, I had to step through the code with a debugger to figure out what the original code was even doing :grimacing: